core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -360,95 +359,27 @@ import scala.util.control.{NoStackTrace, NonFatal}
-              log.warning(
-                "Consumer poll({}) interrupted with WakeupException (#{} of max-wakeups = {}, wakeup-timeout = {}).",
-                timeout,
-                wakeups,
-                settings.maxWakeups,
-                settings.wakeupTimeout.toCoarsest
-              )
-              if (initialPoll) {
-                log.error(
-                  "Initial consumer poll({}) with bootstrap servers {} did not succeed, still trying",
-                  timeout,
-                  settings.properties.getOrElse(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "not set")
-                )
-              }
-            }
-
-            // If the current consumer is using group assignment (i.e. subscriptions is non empty) the wakeup might
-            // have prevented the re-balance callbacks to be called leaving the Source in an inconsistent state w.r.t
-            // assigned TopicPartitions. In order to reconcile the state we manually call callbacks for all added/remove
-            // TopicPartition assignments aligning the Source's state the consumer's.
-            // We are safe to perform the operation here since the poll() thread has been aborted by the wakeup call
-            // and there are no other threads are using the consumer.
-            // Note: in case of manual partition assignment this is not needed since rebalance doesn't take place.
-            if (subscriptions.nonEmpty) {
-              val newAssignments = consumer.assignment().asScala
-              reconcileAssignments(currentAssignmentsJava.asScala.toSet, newAssignments.toSet)
-            }
-          }
-          throw WakeupHandled
-      }
-
     try {
       if (requests.isEmpty) {
         // no outstanding requests so we don't expect any messages back, but we should anyway
         // drive the KafkaConsumer by polling
-
         def checkNoResult(rawResult: ConsumerRecords[K, V]): Unit =
           if (!rawResult.isEmpty)
             throw new IllegalStateException(s"Got ${rawResult.count} unexpected messages")
-
         consumer.pause(currentAssignmentsJava)
-        checkNoResult(tryPoll(0))
+        checkNoResult(consumer.poll(java.time.Duration.ZERO))
 
+        // COMMIT PERFORMANCE OPTIMIZATION
         // For commits we try to avoid blocking poll because a commit normally succeeds after a few
         // poll(0). Using poll(1) will always block for 1 ms, since there are no messages.
         // Therefore we do 10 poll(0) with short 10 μs delay followed by 1 poll(1).
         // If it's still not completed it will be tried again after the scheduled Poll.
         var i = 10
         while (i > 0 && commitsInProgress > 0) {
           LockSupport.parkNanos(10 * 1000)
-          val pollTimeout = if (i == 1) 1L else 0L
-          checkNoResult(tryPoll(pollTimeout))
+          val pollTimeout = if (i == 1) oneMilli else java.time.Duration.ZERO
+          checkNoResult(consumer.poll(pollTimeout))
```

Comment:
We should take measures and try to find out if any `consumer.poll(ZERO)` does anything useful. Maybe `parkNanos(100)` followed by a `consumer.poll(oneMilli)` is just as good.